backend/app/auth/supabase/supabase_auth_service.py

```diff
@@ -0,0 +1,82 @@
+from typing import override
+
+import supabase
+from gotrue.types import AuthResponse
+from supabase import AuthError, AuthInvalidCredentialsError, SupabaseAuthClient
+
+from app.auth import AuthResult, AuthService, UserResult
+from app.auth.supabase import (
+    SupabaseAuthInvalidCredentialsException,
+    SupabaseSignInAnonymouslyResult,
+    SupabaseSignInResult,
+    SupabaseSignOutResult,
+    SupabaseSignUpResult,
+    SupabaseUserException,
+    SupabaseUserResult,
+)
+from app.models import Credentials, Token, settings
+
+
+class SupabaseAuthService(AuthService):
+    def __init__(self):
+        self.__auth_client: SupabaseAuthClient = supabase.create_client(
```

Comment:
Would rather see settings passed in than use a global variable if you can avoid it.